### PR TITLE
Basic LeverageEngine.unwindLeveragedPosition with tests

### PR DESCRIPTION
**Pull request by CoreyMartin**

### CODE REVIEW

File: contracts/LeverageEngine.sol [hunk 1]
```diff
@@ -98,12 +98,11 @@ contract LeverageEngine is ReentrancyGuard, AccessControl {
     /// must check that the msg.sender owns the NFT
     /// provide msg.sender address to coordinator destroy position
     ///
-    /// @param positionId the NFT ID of the position
-    function destroyLeveragedPosition(uint256 positionId) external expectInitialized nonReentrant {
-        require(_positionToken.ownerOf(positionId) == msg.sender, "Caller address does not own this position token");
+    /// @param positionTokenId the NFT ID of the position
+    function unwindLeveragedPosition(uint256 positionTokenId) external expectInitialized nonReentrant {
```
Comment (CoreyMartin) [inline contracts/LeverageEngine.sol:102]: Changed to "unwind" so people don't get confused and think they are destroying their whole position without getting anything back :)

File: contracts/LeverageEngine.sol [hunk 2]
```diff
@@ -98,12 +98,11 @@ contract LeverageEngine is ReentrancyGuard, AccessControl {
...
+        require(_positionToken.ownerOf(positionTokenId) == msg.sender, "Caller address does not own this position token");
         // coordinator needs to change to unwindLeveragedOUSD to assume coordinator as shares owner. Yotam adding task
```
Comment (yotamdaniel) [inline contracts/LeverageEngine.sol:104]: This comment can be removed right?

File: test/LeverageEngine.ts [hunk 3]
```diff
@@ -42,68 +42,106 @@ describe("LeverageEngine test suit", async function () {
...
     });
 
-    describe("createLeveragedPosition", async function () {
-        it("Should revert if cycles is greater than global max cycles", async function () {
-            const maxCycles = await r.parameterStore.getMaxNumberOfCycles();
-            const promise = r.leverageEngine.createLeveragedPosition(ethers.utils.parseEther("1"), maxCycles.add(1));
-            await expect(promise).to.be.revertedWith("Number of cycles must be lower then allowed max");
+    it("Should revert if cycles is greater than global max cycles", async function () {
+        const maxCycles = await r.parameterStore.getMaxNumberOfCycles();
+        const promise = r.leverageEngine.createLeveragedPosition(ethers.utils.parseEther("1"), maxCycles.add(1));
```
Comment (yotamdaniel) [inline test/LeverageEngine.ts:56]: change parse ether to parse units :P

**Review by yotamdaniel — APPROVED** [review]
looks good with some minor comments

File: test/LeverageEngine.ts (hunk 3 above)
Comment (CoreyMartin) [reply test/LeverageEngine.ts]: done!

File: contracts/LeverageEngine.sol (hunk 2 above)
Comment (CoreyMartin) [reply contracts/LeverageEngine.sol]: removed

**Pull request merged**